File: java/exchangeapi/src/main/java/com/thesoftwaregorilla/tdd/money/http/GetRequest.java
package com.thesoftwaregorilla.tdd.money.http;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.logging.Level;
import java.util.logging.Logger;

public class GetRequest {

    private final Logger logger = Logger.getLogger(GetRequest.class.getName());

    public String url;
    public GetRequest(String url) {
        this.url = url;
    }

    public String fetchResponse() {
        try (HttpClient client = HttpClient.newHttpClient()){
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(new URI(url))
                    .GET()
                    .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            return response.body();
        } catch (Exception e) {
            logger.log(Level.SEVERE, "Error fetching response: " + e.getMessage());
            return null;
        }
    }

    public String getUrl() {
        return url;
    }
}

File: cs/exchangeapi/Money/http/GetRequest.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Microsoft.Extensions.Logging;

namespace TheSoftwareGorilla.TDD.Money.Http;

public class GetRequest
{
    public string Url { get; }

    public GetRequest(string url)
    {
        Url = url;

    }

    public async Task<string> FetchResponseAsync()
    {
        using HttpClient client = new HttpClient();
        HttpResponseMessage response = await client.GetAsync(Url);
        response.EnsureSuccessStatusCode();
        return await response.Content.ReadAsStringAsync();
    }


}